mypy/subtypes.py

```diff
@@ -463,8 +463,12 @@ def visit_instance(self, left: Instance) -> bool:
                         assert unpacked.type.fullname == "builtins.tuple"
                         if isinstance(get_proper_type(unpacked.args[0]), AnyType):
                             return not self.proper_subtype
-            # TODO: we need a special case similar to above to consider (something that maps to)
-            # tuple[Any, ...] a subtype of Tuple[<whatever>].
+                if (
+                    left.type.fullname == "builtins.tuple"
```

Comment:
I think this should use `has_base()` + `map_instance_to_supertype()` logic like above. Otherwise this breaks transitivity of subtyping.